Python/sysmodule.c

```diff
@@ -668,6 +668,46 @@ The flag constants are defined in the ctypes and DLFCN modules.");
 
 #endif  /* HAVE_DLOPEN */
 
+static PyObject *
+sys_set_int_max_str_digits(PyObject *self, PyObject *args)
+{
+    int new_val;
+    PyThreadState *tstate = PyThreadState_GET();
+    if (!PyArg_ParseTuple(args, "i:set_int_max_str_digits", &new_val))
+        return NULL;
+    if (!tstate)
+        return NULL;
+    if ((!new_val) || new_val >= (_PY_LONG_MAX_STR_DIGITS_THRESHOLD)) {
+        tstate->interp->int_max_str_digits = new_val;
+    } else {
+        PyErr_Format(
+            PyExc_ValueError, "maxdigits must be 0 or larger than %d",
+            _PY_LONG_MAX_STR_DIGITS_THRESHOLD);
+        return NULL;
+    }
+    Py_INCREF(Py_None);
+    return Py_None;
+}
```

Comment:
The control flow here is a little confusing. If we could assume C99 and put the initialization of `tstate` in the appropriate conditional, that could be clearer.

When `PyArg_ParseTuple` returns, do we guarantee that `new_val` will have a value? I can't tell from skimming the rest of the code if it's safer to set a value of `0` or `NULL` explicitly at declaration.

Reply:
I can see what you mean with the non-local assignment to tstate, but this same control flow is used throughout including over a dozen times just in this file, of setting tstate, then parsing args if any, then checking the value of tstate.
`new_val` will have a value as long as `PyArg_ParseTuple` returns true https://docs.python.org/2.7/c-api/arg.html#c.PyArg_ParseTuple

Reply:
:+1: Better to stick with the local style in that case.